bpf/bpf_lxc.c

```diff
@@ -793,6 +793,13 @@ static __always_inline int handle_ipv4_from_lxc(struct __ctx_buff *ctx,
 		struct egress_info *info;
 		struct endpoint_key key = {};
 
+		/* If tunnel endpoint is found in ipcache, it means the remote endpoint is
+		 * in cluster. In this case, we should skip egress gateway. If destination
+		 * is either remote node or host node, also skip egress gateway.
+		 */
+		if (tunnel_endpoint != 0 || *dstID == REMOTE_NODE_ID || *dstID == HOST_ID)
+			goto skip_egress_gateway;
```

Comment:
This also means we will skip the egress gateway lookup and redirection for local endpoints in case `ENABLE_ROUTING` is undefined, no? If that's okay (I think it is), we should extend the comment to document that behavior as well.

Reply:
hmm. Trying to figure out why it's related to `ENABLE_ROUTING`. `tunnel_endpoint` and `dstID` are coming from the here: 

https://github.com/cilium/cilium/blob/673ccd7858ac09f799db5d76cc44a71c2e2754c5/bpf/bpf_lxc.c#L616-L635

Reply:
Ah, I could have explained this better :-)

The relation to `ENABLE_ROUTING` comes from lines 753--778:
- If `ENABLE_ROUTING` is defined and the destination is a local endpoint, we will deliver the packet via `ipv4_local_delivery` and never reach the egress gateway logic here. That didn't change in this patch.
- If `ENABLE_ROUTING` is undefined, we will execute the logic here. `tunnel_endpoint` is set to 0 for local endpoints. So for those endpoints under `! ENABLE_ROUTING`, we will now skip the egress gateway.